src/arrayMethodJoin.js

```diff
@@ -5,8 +5,23 @@
  *
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function() {
-    // write code here
+  [].__proto__.join2 = function(str = ',') {
+    let string = '';
+    let separator = str;
+    if (separator === null) {
+      separator = 'null';
+    }
+    this.forEach((currentValue, index, array) => {
+      const supplemented = currentValue === undefined || currentValue === null
+        ? ''
+        : currentValue;
```

Comment:
you can replace this ternary operator with 
const supplemented = currentValue || '';